src/field.rs

```diff
@@ -181,6 +181,73 @@ pub fn summator_incompr(
     }
 }
 
+/// The Fourier method for scalar fields.
+///
+/// Computes the periodic, isotropic spatial random field $u(x)$ by the Fourier
+/// method according to
+/// $$
+/// u(x) = \sum_{i=1}^N \sqrt{2S(k_i\Delta k}\left(
+/// z_{1,i} \cdot \cos(\langle k_i, x \rangle) +
+/// z_{2,i} \cdot \sin(\langle k_i, x \rangle)\right)
+/// $$
+/// with
+/// * $S$ being the spectrum of the covariance model
+/// * $N$ being the number of Fourier modes
+/// * $z_1, z_2$ being independent samples from a standard normal distribution
+/// * $k$ being the equidistant Fourier grid
+/// and are given by the argument `modes`.
+/// * $\Delta k$ being the cell size of the Fourier grid
+///
+/// # Arguments
+///
+/// * `spectrum_factor` - the pre-calculated factor $\sqrt{2S(k_i\Delta k)}
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp; dim = Fourier modes $N$
+/// * `modes` - equidistant Fourier grid, $k$ in Eq. above
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = (spatial dim. of field $d$, Fourier modes $N$)
+/// * `z1` - independent samples from a standard normal distribution
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = Fourier modes $N$
+/// * `z2` - independent samples from a standard normal distribution
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = Fourier modes $N$
+/// * `pos` - the position $x$ where the spatial random field is calculated
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = (spatial dim. of field $d$, no. of spatial points where field is calculated $j$)
+/// * `num_threads` - the number of parallel threads used, if None, use rayon's default
+///
+/// # Returns
+///
+/// * `summed_modes` - the isotropic spatial field
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = no. of spatial points where field is calculated $j$
+pub fn summator_fourier(
+    spectrum_factor: ArrayView1<'_, f64>,
+    modes: ArrayView2<'_, f64>,
+    z1: ArrayView1<'_, f64>,
+    z2: ArrayView1<'_, f64>,
+    pos: ArrayView2<'_, f64>,
+    num_threads: Option<usize>,
+) -> Array1<f64> {
+    assert_eq!(modes.dim().0, pos.dim().0);
+    assert_eq!(modes.dim().1, z1.dim());
+    assert_eq!(modes.dim().1, z2.dim());
+
+    rayon::ThreadPoolBuilder::new()
+        .num_threads(num_threads.unwrap_or(rayon::current_num_threads()))
+        .build()
+        .unwrap()
+        .install(|| {
+            Zip::from(pos.columns()).par_map_collect(|pos| {
+                Zip::from(modes.columns())
+                    .and(spectrum_factor)
+                    .and(z1)
+                    .and(z2)
+                    .fold(0.0, |sum, k, &spectrum_factor, &z1, &z2| {
```

Comment:
Since the accumulator is a scalar and hence cheap to copy/reduce, I wonder whether this would actually benefit from using a [`par_fold`](https://docs.rs/ndarray/latest/ndarray/struct.Zip.html#method.par_fold) instead of the sequential fold?

Reply:
And thanks again :-) I'll run a few benchmarks soon to see, if you are right.

Reply:
Interesting, the `par_fold` version is about 50% slower in my benchmarks.

Reply:
Quite possible that the overhead is unhelpful for the inner loop which would otherwise benefit from auto-vectorization. Using [`with_min_len`](https://docs.rs/rayon/latest/rayon/iter/trait.IndexedParallelIterator.html#method.with_min_len) might be a nice compromise between granularity and overhead, but it does add another tunable.